### PR TITLE
Add documentation for IoT logging

### PR DESCRIPTION
**Pull request by ctron**
### Type of change

- Documentation

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/modules/proc-iot-logging.adoc [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  logging:
+    level: info # <1>
+    loggers: # <2>
+      io.enmasse: debug # <3>
+      io.netty: error # <4>
+----
+<1> The default global log level. If omitted, `info` will be used.
+<2> The section for log channel specific entries.
+<3> Lowers the filtering to `debug` level for messages in the channel `io.enmasse`
+<4> Raises the filtering to `error` level for messages in the channel `io.netty`
```
Comment (jbtrystram) [inline documentation/modules/proc-iot-logging.adoc:50]: So the log filter level or log level ? It feels one is the inverse of the other.

File: documentation/modules/proc-iot-logging.adoc [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+          <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
+        </encoder>
+      </appender>
+      <root level="debug">
+        <appender-ref ref="STDOUT" />
+      </root>
+    </configuration>
+----
+<1> The application specific configuration file. This file is generated
+    by the operator. Any changes will be overwritten.
```
Comment (hmaclean73) [inline documentation/modules/proc-iot-logging.adoc:208]:
```suggestion
    and overwrites any changes.
```

File: documentation/modules/proc-iot-logging.adoc [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+<2> The effective logback configuration, applied by the system. Do not
+    change this, as it will be overwritten by the operator.
+<3> The full XML based logback configuration
```
Comment (hmaclean73) [inline documentation/modules/proc-iot-logging.adoc:211]:
```suggestion
<3> The full XML-based logback configuration
```

File: documentation/modules/proc-iot-logging.adoc [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+. The operator will detect changes on the `ConfigMap` resource, apply
```
Comment (hmaclean73) [inline documentation/modules/proc-iot-logging.adoc:212]:
```suggestion
. The operator detects changes on the `ConfigMap` resource, applies
```

File: documentation/modules/proc-iot-logging.adoc [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+  the logging configuration and re-deploy all required components.
```
Comment (hmaclean73) [inline documentation/modules/proc-iot-logging.adoc:213]:
```suggestion
  the logging configuration and re-deploys all required components.
```

File: documentation/modules/con-iot-logging.adoc [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+[id='con-iot-logging-{context}']
+= IoT logging
+
+By default the IoT components of {ProductName} are using a rather conservative
+logging configuration for the infrastructure services. This is done order to
+preserve resources for the normal operation.
+
+{ProductName} also allows to enable operation level tracing using Jaeger,
+for more in depth tracing scenarios.
```
Comment (hmaclean73) [inline documentation/modules/con-iot-logging.adoc:13]:
```suggestion
for more in-depth tracing scenarios.
```

File: documentation/modules/con-iot-logging.adoc [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+* A default log level for all IoT components
+* A default configuration for specific log channels
+* A service specific default log level
+* A service specific configuration for specific log channels
+* A service specific custom log configuration file
+
+All configuration is part of the `IoTConfig` instance and will be applied to the
+services by the {ProductName} operator. The only exception is the service specific
+log configuration file. It can either be provided through the `IoTConfig` resource,
```
Comment (hmaclean73) [inline documentation/modules/con-iot-logging.adoc:29]:
```suggestion
log configuration file, which can either be provided through the `IoTConfig` resource
```

File: documentation/modules/con-iot-logging.adoc [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+The following log levels can be used:
+
+error:: Error conditions. Typically an unexpected condition which may impact the stability of the system.
+warn:: Warning conditions. Typically an expected condition, which may impact the current operation or stability of the system.
+info:: Informational messages. Typically an expected event, which may impact the current operation.
+debug:: Messages for debugging.
+trace:: All messages.
+
+When limiting output by log level, lower levels will always include
```
[the remark this answers is not in the document]
Comment (ctron) [reply documentation/modules/con-iot-logging.adoc:42]: I don't think so: "info" is lower than "error". However setting the level to "info" includes "error" messages.

[the remark this answers is not in the document]
Comment (jbtrystram) [reply documentation/modules/con-iot-logging.adoc:42]: Got it. i got the terminology inverted and was confused. That said it goes a bit against the idea of "increasing the log verbosity" which intuitively would have me select a "higher" log level.

File: documentation/modules/proc-iot-logging.adoc [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+
+== Configure the global log levels
+
+The global logging configuration is applied to all
+services which do not have any explicit logging configuration.
+
+By default the global log level is `info`.
+
+.Prerequisites
+* link:{BookUrlBase}{BaseProductVersion}{BookNameUrl}#iot-installing-services-{context}[The IoT services are installed].
```
[the remark this answers is not in the document]
Comment (ctron) [reply documentation/modules/proc-iot-logging.adoc:21]: I don't really know what to do here, and I never really understood how this all works. It seems the section is defined as `[id='iot-installing-services-{context}']` in `proc-iot-installing-services.adoc`. What would be the proper replacement, when not using `{context}`? Should the section ID (and all links to it) be changed as well?

**Comment by ctron** [comment]
Thanks for all the suggestions! :grinning: All are applied. I am just not sure what to do with `{context}`.

File: documentation/modules/proc-iot-logging.adoc (hunk 9 above)
[the remark this answers is not in the document]
Comment (hmaclean73) [reply documentation/modules/proc-iot-logging.adoc:21]: I think in this case, I can try fixing it in the upstream docs (if the link breaks) once it is merged.

**Pull request merged**